### PR TITLE
Credential card rework.

### PR DESCRIPTION
**Pull request by JoshDunnDev**

### CODE REVIEW

File: CredentialCardDetail.vue [hunk 1]
```diff
@@ -0,0 +1,174 @@
+<template>
+  <q-card class="row br-credential-card">
+    <q-card-title class="text-center s-card-title">
+      Credential Details
+    </q-card-title>
+    <div class="row justify-between s-card-info">
+      <div class="column items-center s-card-info-left">
+        <q-card-media>
+          <q-icon
+            v-if="useDefaultImage || !credential.issuerLogo & !credential.image"
```
Comment (mattcollier) [inline CredentialCardDetail.vue]: I don't think this does the right thing, see: https://repl.it/repls/GuiltySpotlessScans

File: CredentialCardDetail.vue [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+              :key="value.id"/>
+          </q-list>
+        </q-card-actions>
+      </div>
+    </div>
+  </q-card>
+</template>
+<script>
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
```
Comment (mattcollier) [inline CredentialCardDetail.vue]: Date here and throughout.

File: credentialMixin.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  },
+  data() {
+    return {
+      showFieldValues: false,
+      useDefaultImage: false
+    };
+  },
+  computed: {
+    fields() {
+      if(!this.credential || !this.schema) {
```
Comment (mattcollier) [inline credentialMixin.js]: preferred: `if(!(a && b))`

File: credentialMixin.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  },
+  methods: {
+    imageError() {
+      this.useDefaultImage = true;
+    }
+  }
+};
+
+function _createFields(fields, source, schema) {
+  console.log('SOURCE', source);
```
Comment (mattcollier) [inline credentialMixin.js]: remove?

File: credentialMixin.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  for(const key in source) {
+    // naively recurse into objects
+    if(typeof source[key] === 'object') {
```
Comment (mattcollier) [inline credentialMixin.js:89]: Does this behave as expected if `source[key]` is an array (which also has typeof object).

Comment (mattcollier) [reply credentialMixin.js:89]: Seems like this sort of algo has been developed a number of times, maybe there's something we can use?
@gannan08 @dlongley @davidlehn

**Review by mattcollier — CHANGES_REQUESTED** [review]
done with first pass.

File: CredentialCardDetail.vue (hunk 1 above)
Comment (JoshDunnDev) [reply CredentialCardDetail.vue]: Updated.

File: credentialMixin.js (hunk 3 above)
Comment (JoshDunnDev) [reply credentialMixin.js]: Updated.

File: CredentialCardDetail.vue (hunk 2 above)
Comment (JoshDunnDev) [reply CredentialCardDetail.vue]: Updated.

File: credentialMixin.js (hunk 4 above)
Comment (JoshDunnDev) [reply credentialMixin.js]: Removed.

**Comment by JoshDunnDev** [comment]
@mattcollier Using `!(credential.issuerLogo && credential.image)` appears to not be working the same as `!credential.issuerLogo && !credential.image`.

**Review by mattcollier — APPROVED** [review]
Only open question is related to recursive `_createFields` helper which I will leave to others to sort out.

File: CredentialCardList.vue [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+          <img
+            v-else
+            :src="credential.issuerLogo"
+            @error="imageError">
+        </q-card-media>
+      </div>
+      <q-card-separator />
+    </div>
+
+    <div v-if="showFieldValues === true">
```
Comment (davidlehn) [inline CredentialCardList.vue]: Can this just test for truthiness?
```suggestion
    <div v-if="showFieldValues">
```

**Comment by gannan08** [comment]
Closed in preference of https://github.com/digitalbazaar/bedrock-vue-credential-card/pull/5.

**Pull request closed without merging**